### ISSUE TITLE
Unable to delete pid file when not running as root

### ISSUE THREAD

**Issue by realsimix** [graygnuorg/pound#10]
Hi,

I see the following issue when running pound as user pound:

> Apr  3 17:14:16 ns1 pound[4066]: got signal 15
> Apr  3 17:14:16 ns1 pound[4069]: shutting down...
> Apr  3 17:14:16 ns1 pound[4069]: can't remove file /var/run/pound/ctl_socket: Permission denied
> Apr  3 17:14:16 ns1 pound[4069]: can't remove file /var/run/pound.pid: Permission denied
> Apr  3 17:14:16 ns1 pound[4066]: got signal 17

I understand that this is because the files are owned by root and in fact it doesn't happen when pound is running as root.

Is this the expected behavior and the error messages should just be ignored?

Thanks,
Simon

**Comment by graygnuorg** [comment]
Yes, it is expected behavior in this case. To eliminate it, I suggest creating a directory for storing pound pid and control socket files, and changing its ownership to the user pound is running as, e.g.:

**Comment by realsimix** [comment]
Thanks for the hint, all working well now.

I'm still wondering something: what is the expected way to handle permissions on the control socket?
I understand that ownership or access mode of the socket has to be changed in order to allow some group of non root users to access pound via poundctl.
Is there a way to let pound create the socket with different permissions?

Thanks,
Simon

**Comment by graygnuorg** [comment]
Pound is pretty minimalistic in this regard right now: the socket is always owned by root and has mode 0700. I plan to add a configuration directive that will allow for specifying socket permissions for the socket. I'll update this issue when it is ready. Please stay tuned.

**Comment by realsimix** [comment]
Sounds very nice, thank you.

**Comment by graygnuorg** [comment]
Hi Simon,

I implemented that in commit 1a348505e0b067ee9e968b0623e53bba5fb7a306.  Please, pull.  See https://www.gnu.org.ua/software/pound/pound.html#Control_socket for detailed discussion. To make the long story short,  `Control` statement can now be used in block form, which allows for specifying additional parameters. For example, to change mode of the socket file:

```
Control
     Socket "/run/pound/pound.sock"
     Mode 660
End
```

**Comment by realsimix** [comment]
Hi Sergey,

Thanks for the new feature, that's exactly what I was looking for.
I'll update my RPM packages and see how it works.

**Comment by realsimix** [comment]
A small detail showed up while testing: detection of the socket in poundctl doesn't work anymore now if the new form using Socket is used:

```
$ poundctl
poundctl: can't determine control socket name; use the -s option
```

Not a big deal because using poundctl -s works. It would still be nice to have this fixed :)

**Comment by graygnuorg** [comment]
Thanks for spotting that. Fixed in e696dbca3addd26fc2328c23c2ed197243a4415f.
Please, pull.

**Comment by realsimix** [comment]
Thanks for the update, it's working perfectly now!

**Comment by graygnuorg** [comment]
Great. I'm closing this issue, then. Thank you.
